### PR TITLE
enhance: support groupby based on scalar-index(#29965)

### PR DESCRIPTION
**Pull request by MrPresent-Han**
related: #29965

### CODE REVIEW

File: internal/core/src/segcore/Reduce.cpp [hunk 1]
```diff
@@ -541,8 +541,7 @@ ReduceHelper::AssembleGroupByValues(
             case DataType::VARCHAR: {
                 auto field_data = group_by_values_field->mutable_string_data();
                 for (std::size_t idx = 0; idx < group_by_val_size; idx++) {
-                    std::string_view val =
-                        std::get<std::string_view>(group_by_vals[idx]);
+                    std::string val = std::get<std::string>(group_by_vals[idx]);
                     *(field_data->mutable_data()->Add()) = val;
```
Comment (yah01) [inline internal/core/src/segcore/Reduce.cpp:545]: Move to avoid copying

File: internal/core/unittest/test_group_by.cpp [hunk 2]
```diff
@@ -337,12 +337,11 @@ TEST(GroupBY, Normal2) {
...
         float lastDistance = 0.0;
         for (size_t i = 0; i < size; i++) {
-            if (std::holds_alternative<std::string_view>(group_by_values[i])) {
-                std::string_view g_val =
-                    std::get<std::string_view>(group_by_values[i]);
+            if (std::holds_alternative<std::string>(group_by_values[i])) {
+                std::string g_val = std::get<std::string>(group_by_values[i]);
                 ASSERT_FALSE(strs_set.count(g_val) >
                              0);  //no repetition on groupBy field
                 strs_set.insert(g_val);
```
Comment (yah01) [inline internal/core/unittest/test_group_by.cpp:347]: Move to avoid copying

**Review by yah01 — CHANGES_REQUESTED** [review]

File: internal/core/src/segcore/Reduce.cpp (hunk 1 above)
Comment (MrPresent-Han) [reply internal/core/src/segcore/Reduce.cpp:546]: changed

File: internal/core/unittest/test_group_by.cpp (hunk 2 above)
Comment (MrPresent-Han) [reply internal/core/unittest/test_group_by.cpp:348]: changed

**Review by yah01 — COMMENTED** [review]
/lgtm

**Pull request merged**